### PR TITLE
refactor(helpers)!: remove v8 deprecated helpers methods

### PR DESCRIPTION
**Pull request by Shinigami92**

### CODE REVIEW

**Review by ST-DDT — DISMISSED** [review]

File: docs/guide/upgrading_v9/2729.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+### Remove deprecated helpers methods
+
+Removed deprecated helpers methods
+
+| old                                     | replacement                                                    |
+| --------------------------------------- | -------------------------------------------------------------- |
+| `faker.helpers.replaceSymbolWithNumber` | `string.replace(/#+/g, (m) => faker.string.numeric(m.length))` |
+| `faker.helpers.regexpStyleStringParse`  | `faker.helpers.fromRegExp`                                     |
```
Comment (matthewmayer) [inline docs/guide/upgrading_v9/2729.md:8]: as this is not an exact replacement i think we should link back to https://v8.fakerjs.dev/guide/upgrading.html#faker-helpers-regexpstylestringparse-deprecated-in-favor-of-faker-helpers-fromregexp or repeat the explanation from there

File: docs/guide/upgrading_v9/2729.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+### Remove deprecated helpers methods
+
+Removed deprecated helpers methods
+
+| old                                     | replacement                                                    |
+| --------------------------------------- | -------------------------------------------------------------- |
+| `faker.helpers.replaceSymbolWithNumber` | `string.replace(/#+/g, (m) => faker.string.numeric(m.length))` |
```
Comment (matthewmayer) [inline docs/guide/upgrading_v9/2729.md:7]: similarly as this is not an exact replacement i think we should give a longer explanation that handles the ! case as well as the #

**Review by matthewmayer — CHANGES_REQUESTED** [review]

File: docs/guide/upgrading_v9/2729.md (hunk 2 above)
Comment (Shinigami92) [reply docs/guide/upgrading_v9/2729.md:7]: can you please take this over and edit with your change requests?

Comment (matthewmayer) [reply docs/guide/upgrading_v9/2729.md:7]: done

File: docs/guide/upgrading_v9/2729.md (hunk 1 above)
Comment (matthewmayer) [reply docs/guide/upgrading_v9/2729.md:8]: done

**Review by ST-DDT — APPROVED** [review]

**Review by matthewmayer — APPROVED** [review]

**Pull request merged**